public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx

```diff
@@ -0,0 +1,326 @@
+  };
+
+  getRenameReplacement = (renameReplacement: string) => {
+    this.setState({ renameReplacement: renameReplacement });
+  };
+
+  onToggle = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.id === "restore_specific_indices") {
+      this.setState({ restoreSpecific: true, snapshot: _.set(this.state.snapshot!, e.target.id, e.target.checked) });
+      return;
+    }
+    if (e.target.id === "restore_all_indices") {
+      this.setState({ restoreSpecific: false, snapshot: _.set(this.state.snapshot!, e.target.id, e.target.checked) });
+      return;
+    }
+    if (e.target.name === "rename_option") {
+      this.setState({ renameIndices: e.target.id, snapshot: _.set(this.state.snapshot!, e.target.id, e.target.checked) });
+      return;
+    }
+
+    this.setState({ snapshot: _.set(this.state.snapshot!, e.target.id, e.target.checked) });
+  };
+
+  render() {
+    const { onCloseFlyout } = this.props;
+    const { indexOptions, selectedIndexOptions, selectedRepoValue, restoreSpecific, snapshot, renameIndices } = this.state;
+
+    return (
+      <EuiFlyout ownFocus={false} onClose={onCloseFlyout} size="m" hideCloseButton>
+        <EuiFlyoutHeader hasBorder>
+          <EuiTitle size="m">
+            <h2 id="flyoutTitle">Restore snapshot</h2>
+          </EuiTitle>
+        </EuiFlyoutHeader>
+
+        <EuiFlyoutBody>
+          <EuiFlexGroup>
+            <EuiFlexItem>
+              <CustomLabel title="Snapshot name" />
+              <h3>{snapshot?.snapshot}</h3>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <CustomLabel title="Status" />
+              <h3>{snapshot?.state}</h3>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <CustomLabel title="Indices" />
+              <a onClick={this.onClickIndices}>{snapshot?.indices.length}</a>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+
+          <EuiSpacer size="xxl" />
+
+          <SnapshotRestoreOption
+            restoreAllIndices={!restoreSpecific}
+            onRestoreAllIndicesToggle={this.onToggle}
+            restoreSpecificIndices={restoreSpecific}
+            onRestoreSpecificIndicesToggle={this.onToggle}
+            width="200%"
+          />
+
+          <EuiSpacer size="l" />
+
+          {restoreSpecific && (
+            <SnapshotIndicesInput
+              indexOptions={indexOptions}
+              selectedIndexOptions={selectedIndexOptions}
+              onIndicesSelectionChange={this.onIndicesSelectionChange}
+              getIndexOptions={this.getIndexOptions}
+              onCreateOption={this.onCreateOption}
+              selectedRepoValue={selectedRepoValue}
+              isClearable={true}
+            />
+          )}
+
+          <EuiSpacer size="l" />
+
+          <SnapshotRenameOptions
+            doNotRename={renameIndices === "do_not_rename"}
+            onDoNotRenameToggle={this.onToggle}
+            addPrefix={renameIndices === "add_prefix"}
```

Comment:
It might be helpful to have options strings like `add_prefix` in an enum.

Reply:
Implemented and pushed